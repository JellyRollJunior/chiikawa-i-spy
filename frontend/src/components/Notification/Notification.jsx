import styles from './Notification.module.css';
import shared from '../../styles/shared.module.css';

const Notification = ({ message, positionStyling = {left: 0, right: 0}, successNotification = true }) => {
  return (
    <h3
      style={positionStyling}
      className={`${styles.notification} ${styles.fadeInfadeOut} ${
        successNotification
          ? `${shared.card} ${styles.successNotification}`
          : shared.primaryButton
      }`}
    >
      {message}
    </h3>
  );
};

export { Notification };
