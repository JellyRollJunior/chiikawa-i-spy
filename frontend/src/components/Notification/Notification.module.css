.notification {
  position: absolute;
  top: 24px;
  width: fit-content;
  min-width: 100px;
  padding: 2px 24px;
  margin-inline: auto;

  text-align: center;
  border-width: 3px; 
}

.successNotification {
  color: var(--dark-green);
}

/* animation */
.fadeInfadeOut {
  animation: fade-in-fade-out 1.5s ease-out;
}

@keyframes fade-in-fade-out {
  from {
    opacity: 0;
  }
  15% {
    opacity: 100;
  }
  85% {
    opacity: 100;
  }
  to {
    opacity: 0;
  }
}
