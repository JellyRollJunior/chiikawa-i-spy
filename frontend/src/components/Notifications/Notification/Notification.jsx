import { useContext, useState } from 'react';
import { NotificationContext } from '../../../providers/notificationContext.jsx';
import styles from './Notification.module.css';
import shared from '../../../styles/shared.module.css';

const Notification = ({ id, message, isError, isTimed }) => {
  const { removeNotification } = useContext(NotificationContext);
  const [isFadingOut, setIsFadingOut] = useState(false);

  return (
    <h3
      key={id}
      className={`${
        isError
          ? `${shared.primaryButton} ${shared.fadeIn}`
          : `${shared.card} ${styles.timedNotification}`
      } 
      ${isTimed && `${shared.fadeInFadeOut}`} 
      ${isFadingOut && `${shared.fadeOut}`}`}
    >
      {message}
      {isError && !isTimed && (
        <button
          className={styles.closeButton}
          onClick={async () => {
            setIsFadingOut(true);
            await new Promise((resolve) => setTimeout(resolve, 250));
            removeNotification(id);
          }}
        >
          Ã—
        </button>
      )}
    </h3>
  );
};

export { Notification };
