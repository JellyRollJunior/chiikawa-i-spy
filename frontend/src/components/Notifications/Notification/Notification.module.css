.timedNotification {
  min-width: 100px;
  padding: 2px 12px;
  border-width: 3px;
  border-radius: 8px;
  color: var(--dark-green);
}

.closeButton {
  padding: 0 0 0 3px;
  color: inherit;
  background-color: inherit;
  border: none;
}
