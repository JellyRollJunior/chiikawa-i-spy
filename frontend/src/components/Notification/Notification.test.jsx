import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Notification } from './Notification.jsx';

describe('Notification component', () => {
  it('renders header with message and success styling by default', () => {
    const { container } = render(<Notification message="Hello" />);
    
    expect(container).toMatchSnapshot();
  });
});
