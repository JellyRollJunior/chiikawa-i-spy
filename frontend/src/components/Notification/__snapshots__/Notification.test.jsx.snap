// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Notification component > renders header with message and success styling by default 1`] = `
<div>
  <h3
    class="_notification_dfd14c _card_5890fc _successNotification_dfd14c"
  >
    Hello
  </h3>
</div>
`;
